lib/context_adapter_communication.js

```diff
@@ -0,0 +1,677 @@
+    },
+    json: true,
+    body: payload
+  };
+  request(requestOptions, callback);
+}
+
+/**
+ * Sends a updateContext request to the configured Context Broker
+ * @param {string} service Service header to be used in the request
+ * @param {string} servicePath Service Path header to be used in the request
+ * @param {object} payload Payload to be included in the request
+ * @param {function} callback Callback to be called informing of any errors or returning
+ *  the obtained response
+ */
+function sendUpdateContext(service, servicePath, payload, callback) {
+  var context = {
+    corr: caHelper.getUnicaCorrelator(),
+    trans: caHelper.getTransactionId(),
+    op: caHelper.getOperationType()
+  };
+
+  caLogger.debug(
+    context,
+    'Sending updateContext to CB at ' + 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH +
+    '/updateContext' + ' with payload: ' + JSON.stringify(payload)
+  );
+
+  var requestOptions = {
+    method: 'POST',
+    uri: 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH + '/updateContext',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Fiware-Service': service || caConfig.SERVICE,
+      'Fiware-ServicePath': servicePath || caConfig.SERVICE_PATH
+    },
+    json: true,
+    body: payload
+  };
+  request(requestOptions, callback);
+}
+
+/**
+ * Checks that an updateContext request sent to the Context Adapter is well-formed
+ * @param {object} request The request sent to the Context Adapter
+ * @param {function} callback Callback to be called with the result of the check
+ * @return {*} boolean True if the updateContext request is valid. False otherwise
+ */
+function checkUpdateContextRequest(request, callback) {
+  var updateContextJSONSchema = {
+    properties: {
+      payload: {
+        id: 'payload',
+        type: 'object',
+        required: true,
+        properties: {
+          contextElements: {
+            id: 'contextElements',
+            type: 'array',
+            required: true,
+            minItems: 0,
+            maxItems: 2,
+            items: {
+              id: '0',
+              type: 'object',
+              required: true,
+              properties: {
+                id: {
+                  id: 'id',
+                  type: 'string',
+                  required: true
+                },
+                type: {
+                  id: 'type',
+                  type: 'string',
+                  required: true
+                },
+                isPattern: {
+                  id: 'isPattern',
+                  type: 'string',
+                  required: true,
+                  pattern: 'false'
+                },
+                attributes: {
+                  id: 'attributes',
+                  type: 'array',
+                  required: true,
+                  minItems: 4,
+                  maxItems: 6,
+                  conform: function(attributes) {
```

Comment:
Fixed in https://github.com/telefonicaid/context-adapter/commit/4e6c67511f36722f78cc717a7c5ae97b88c5ab54